examples/scene-example.ts

```diff
@@ -0,0 +1,378 @@
+
+    protected _camera: Camera;
+
+    protected _scene: SceneNode;
+
+    protected _texture: Texture2D;
+    protected _framebuffer: Framebuffer;
+
+    /**
+     * Initializes and sets up rendering passes, navigation, loads a font face and links shaders with program.
+     * @param context - valid context to create the object for.
+     * @param identifier - meaningful name for identification of this instance.
+     * @param mouseEventProvider - required for mouse interaction
+     * @returns - whether initialization was successful
+     */
+    protected onInitialize(context: Context, callback: Invalidate,
+        mouseEventProvider: MouseEventProvider,
+        /* keyEventProvider: KeyEventProvider, */
+        /* touchEventProvider: TouchEventProvider */): boolean {
+
+        this._framebuffer = new DefaultFramebuffer(this._context, 'DefaultFBO');
+        this._framebuffer.initialize();
+
+        /* Create and configure camera. */
+
+        this._camera = new Camera();
+        this._camera.center = vec3.fromValues(0.0, 0.0, 0.0);
+        this._camera.up = vec3.fromValues(0.0, 1.0, 0.0);
+        this._camera.eye = vec3.fromValues(0.0, 0.0, 3.0);
+        this._camera.near = 0.1;
+        this._camera.far = 8.0;
+
+        /* Create and configure navigation */
+
+        this._navigation = new Navigation(callback, mouseEventProvider);
+        this._navigation.camera = this._camera;
+
+        /* Create and configure forward pass. */
+
+        this._forwardPass = new ForwardSceneRenderPass(context);
+        this._forwardPass.initialize();
+
+        this._forwardPass.camera = this._camera;
+        this._forwardPass.target = this._framebuffer;
+
+        /* Create scene. */
+
+        this.generateScene();
+        this._forwardPass.scene = this._scene;
+
+        return true;
+    }
+
+    /**
+     * Uninitializes Buffers, Textures, and Program.
+     */
+    protected onUninitialize(): void {
+        super.uninitialize();
+
+        // TODO: make sure that all meshes and programs inside of the scene get cleaned
+
+        // this._mesh.uninitialize();
+        // this._meshProgram.uninitialize();
+    }
+
+    /**
+     * This is invoked in order to check if rendering of a frame is required by means of implementation specific
+     * evaluation (e.g., lazy non continuous rendering). Regardless of the return value a new frame (preparation,
+     * frame, swap) might be invoked anyway, e.g., when update is forced or canvas or context properties have
+     * changed or the renderer was invalidated @see{@link invalidate}.
+     * Updates the navigaten and the AntiAliasingKernel.
+     * @returns whether to redraw
+     */
+    protected onUpdate(): boolean {
+        const gl = this.context.gl;
+        // Resize
+        if (this._altered.frameSize) {
+            this._camera.viewport = [this._frameSize[0], this._frameSize[1]];
+            gl.viewport(0, 0, this._frameSize[0], this._frameSize[1]);
```

Comment:
I'm not sure if a render pass should rely on the global OpenGL viewport. I think it is more stable to assure the correct size before each draw call.